python/interpret_community/common/explanation_utils.py

```diff
@@ -96,14 +99,30 @@ def _get_raw_feature_importances(importance_values, raw_to_output_feature_maps):
         if len(importance_values.shape) > 2:
             raw_importances = _multiply_sparse_matrix_3d_numpy_tensor(importance_values, raw_to_output_map.T)
         else:
-            raw_importances = raw_to_output_map.dot(importance_values.T).T
+            if _is_identity(raw_to_output_map):
+                raw_importances = importance_values
+            else:
+                raw_importances = raw_to_output_map.dot(importance_values.T).T
             if issparse(raw_importances):
                 raw_importances = raw_importances.toarray()
     else:
         raw_importances = importance_values.dot(raw_to_output_map.T)
     return raw_importances.squeeze(0) if orig_single_dimensional_importances else raw_importances
 
 
+def _is_identity(matrix):
+    """Checks if the given sparse matrix is identity matrix.
```

Comment:
Can you add a unit test for this function?

Reply:
If this is only intended to be called for sparse matrix, then maybe check if the matrix is sparse before referencing the nnz property.

Reply:
done, added check if matrix is sparse before using nnz

Reply:
done, added a unit test as well